### ISSUE TITLE
Training loss weighting makes the identity map a free lunch

### ISSUE THREAD

**Issue by nikhilr612** [nikhilr612/tinyflow#6]
## Summary

The CFM objective is correct, but under the `x_hat` parameterization it becomes an
x-prediction MSE weighted by `(1-t)^-2`, and under that weighting the identity map
`x_hat = x_t` is a uniform free lunch. The model learns it and stops.

## Detail

`ImageFM.train_step` derives velocity as `u = (x_hat - x_t)/max(1-t, 1e-5)` and regresses it
onto `x_1 - x_0`. Since `x_t = t*x_1 + (1-t)*x_0` gives `x_1 - x_t = (1-t)(x_1-x_0)` exactly,
the residual is `u - (x_1-x_0) = (x_hat - x_1)/(1-t)`, so in terms of what the network emits
the loss is identically

```
0.5 * mean( (x_hat - x_1)^2 / (1-t)^2 )
```

At the optimum this is bounded — `Var(x_1|x_t)` shrinks like `(1-t)^2` and cancels the weight.
The problem is what else satisfies it. Because `x_t - x_1 = (1-t)(x_0-x_1)`, the trivial map
`x_hat = x_t` caps the weighted loss at `0.5*E(x_0-x_1)^2 = 0.657` **uniformly in t**, and the
skip connections make that map trivially easy to learn.

| t | `x_hat=0` (init) | `x_hat=x_t` (identity) | optimal `E[x_1\|x_t]` |
|---|---|---|---|
| 0.0 | 0.16 | 0.657 | 0.157 |
| 0.9 | 15.7 | 0.657 | 0.594 |
| 0.99 | 1568 | 0.657 | 0.510 |
| 0.999 | 156800 | 0.657 | 0.501 |

Under `t ~ U[0,1]` the top 0.1% of samples carry 99.5% of the loss mass and the bottom 90%
carry 0.00%. Once the identity is learned those dominant samples are already near-optimal, so
almost no gradient reaches low `t` — which is where the actual generative modelling happens
(`t=0` is noise).

## Evidence from the last run

- Epochs 0-3 collapse `1943 -> 545 -> 208 -> 16.7` as it learns the identity.
- Minimum epoch loss over the whole 415-epoch run: **0.753**, against the predicted identity
  plateau of **0.657**. It parked there and never left.
- With `x_hat ~ x_t`, velocity `~ 0`, so the ODE barely moves `x_0` and the sample is the
  input Gaussian. Measured: `runs/sample_final.png` has neighbour-difference energy 79.5 vs
  26.0 for real data.

## Interaction with #7

These two compound. `argmin E_t[w(t)*||f(x_t,t)-x_1||^2]` decomposes over `t`, so with good
time conditioning each slice is an independent least-squares problem and `w(t)` cancels. But a
**t-blind** model must commit to one function of `x_t`, and solving for the best linear gain
`x_hat = a*x_t` in closed form gives:

| weighting | optimal t-blind gain |
|---|---|
| `w=(1-t)^-2` (current) | **a* = 1.0000** (the identity) |
| `w=1` | a* = 0.359 |

So the near-constant time embedding is what makes the identity the *global optimum* of the
model class the network can currently express.

Fixing the time embedding alone is not sufficient, though: the trunk convs are shared across
all `t` (only FiLM is t-specific) and receive the weighted *sum* of per-sample gradients. In a
typical batch of 128 the max weight is ~13,700 against a median of 3.4, and the whole `t<0.5`
half contributes 0.5% of the gradient — so low-`t` still learns ~200x too slowly, and
`clip_by_global_norm(1.0)` renormalises the sum so the step direction is set by top-`t` samples
anyway.

## Proposed fix

Reweight to `w(t) = 1`, i.e. train directly on x-prediction MSE. This is a **reweighting, not a
correction of a wrong objective**: the minimizer is the posterior mean `E[x_1|x_t,t]` for any
strictly positive `w(t)`, so the optimum is unchanged while the identity stops being free and
gradient mass lands where the modelling is. Same move as DDPM's `L_simple`.

Keep the `x_hat` parameterization (it is what enables the Sobel edge loss). Sampling needs no
change of formula — the network still estimates `E[x_1|x_t]`, so the velocity is still
`(x_hat - x)/(1-t)`.

Note the one real cost: in velocity space `w=1` corresponds to weight `(1-t)^2`, which
under-weights late `t` where fine detail is decided. The t-stratified validation loss
(#14) is what exposes that; the escape hatch is the `t`-sampling distribution,
not the weight.

Also: `edge_weight=1.0` was previously irrelevant (drowned by a term ~1e5 larger) and becomes
live after this change. Lower the default and expose a flag.

**Comment by nikhilr612** [comment]
Fixed by `train_step` in models/imagefm.py (commit 7ad2eed, "fix: loss weighting, time embedding, GroupNorm, adaLN-Zero, shared sampler"). The loss is now denom-floored velocity-space x-prediction per the JiT recipe, so the identity map is no longer a free-lunch minimum. Current code confirmed correct.
